miro_preprocessor/image_sorter/src/sorter_logic.py

```diff
@@ -6,23 +6,206 @@
 dictionary and returns an instance of ``Decision``.
 """
 
+import dateutil.parser
 import enum
+import re
+
+
+class Rules:
+    def __init__(self, collection, image_data):
+        self.collection = collection
+        self.image_data = image_data
+
+    _empty_title_strings = [
+        "-",
+        "--",
+        "Awaiting description",
+        "Awaiting removal from MIRO as duplicate",
+        "Awaiting captions and consent form",
+        "Awaiting catalogue details",
+        "Awaiting caption info",
+        "No info available about this object",
+        "No neg"
+    ]
+
+    _cc_accesses = [
+        "CC-0",
+        "CC-BY",
+        "CC-BY-NC",
+        "CC-BY-NC-ND",
+    ]
+
+    @staticmethod
+    def _normalise_string(s):
+        if s is not None:
+            s = s.lower().strip()
+        if s == "":
+            s = None
+        return s
+
+    def _get(self, key):
+        return self.image_data.get(key)
+
+    def _get_normalised(self, key):
+        return self._normalise_string(self._get(key))
+
+    def _compare(self, key, value):
+        return self._get_normalised(key) == self._normalise_string(value)
+
+    def _is_collection(self, collection_name):
+        return self._normalise_string(self.collection) == self._normalise_string(f"images-{collection_name}")
+
+    def _search(self, regex, key):
+        return re.search(regex, self._get_normalised(key))
+
+    def _is_blank(self, key):
+        return self._get_normalised(key) is None
+
+    def _key_matches(self, key, string_list):
+        return any([self._compare(key, string) for string in string_list])
+
+    def _parse_date(self, date_string):
+        return dateutil.parser.parse(date_string, dayfirst=True)
+
+    def is_collection(self, *args):
+        return any([self._is_collection(collection) for collection in args])
```

Comment:
If you drop the square brackets, `any()` will lazily generate the conditions, and stop as soon as it finds a truthy condition. This is slightly more efficient.

(Why: Python has _generator comprehensions_, which are like list comprehensions, but lazily evaluated. e.g.

```python
my_generator = (self._is_collection(collection) for collection in args)
```

There's then a bit of syntactic sugar that says if the only argument to a function is a generator, you can drop one pair of parentheses.)

Reply:
Also, nice use of `*args`! :)